src/spi.rs

```diff
@@ -326,54 +340,63 @@ impl<'d> SpiBus for SpiBusMasterDriver<'d> {
     }
 }
 
-/// Master SPI abstraction
 pub struct SpiMasterDriver<'d> {
-    host: u8,
-    device: spi_device_handle_t,
-    max_transfer_size: usize,
-    _p: PhantomData<&'d mut ()>,
+    pub(crate) handle: spi_host_device_t,
+    pub(crate) lock: CriticalSection,
+    pub(crate) max_transfer_size: usize,
+    _p: PhantomData<&'d ()>,
 }
 
 impl<'d> SpiMasterDriver<'d> {
     /// Create new instance of SPI controller for SPI1
     ///
     /// SPI1 can only use fixed pin for SCLK, SDO and SDI as they are shared with SPI0.
-    pub fn new_spi1(
-        spi: impl Peripheral<P = SPI1> + 'd,
+    pub fn new_spi1<SPI: Spi>(
+        _spi: impl Peripheral<P = SPI1> + 'd,
         sclk: impl Peripheral<P = gpio::Gpio6> + 'd,
         sdo: impl Peripheral<P = gpio::Gpio7> + 'd,
         sdi: Option<impl Peripheral<P = gpio::Gpio8> + 'd>,
-        cs: Option<impl Peripheral<P = impl OutputPin> + 'd>,
-        config: &config::Config,
+        dma: Option<Dma>,
```

Comment:
No need for `Option`. `Dma` enum has `Dma::Disabled` entry.

Reply:
yeah that was a bit of a lazy addition, so for people who dont care  just can fill the constructor with None :D